### PR TITLE
Add and show success page after sending in a successful application

### PR DESCRIPTION
**Pull request by johannth**
Adds a final step to the grant submission form by moving the user to `/grant/success`.

screenshot 2017-03-16 11 59 18

### CODE REVIEW

File: src/actions/grant.js [hunk 1]
```diff
@@ -127,16 +138,19 @@ export function submitGrant() {
         dispatch(showValidationErrors({ form: errorMessages }));
         return false;
       }
+
+      const submissionId = data.createSubmission.id;
+      log('success!');
+      dispatch(submitGrantSuccess(submissionId));
+      history.push('/grant/success');
```
Comment (johannth) [inline src/actions/grant.js]: Not sure if this is the right place for this. Feels cleaner than having it in the reducer. I love how elm does this with commands returned in the reducer. Much nicer than the redux pattern.

File: src/routes/grantform/GrantFormSuccess.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import Link from '../../components/Link';
+import history from '../../core/history';
+
+const GrantFormSuccess = ({ submission }) => {
+  if (!submission) {
+    if (history) {
+      history.push('/');
+    }
+    return null;
+  }
```
Comment (johannth) [inline src/routes/grantform/GrantFormSuccess.js]: This is a hack. Should probably be done in the route but I couldn't figure out how.

File: src/routes/grantform/GrantFormSuccess.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  const steps = [
+    `We've sent an email to ${submission.email} that contains this information.`,
+    'Our team has received the details of your application and will begin voting as soon as possible.',
```
Comment (johannth) [inline src/routes/grantform/GrantFormSuccess.js]: Once we add the permalink page we should add a link to it here.

File: src/actions/grant.js (hunk 1 above)
Comment (kristjanmik) [reply src/actions/grant.js]: I will make some changes that might shine some light on solutions for what you are thinking

File: src/routes/grantform/GrantFormSuccess.js (hunk 2 above)
Comment (kristjanmik) [reply src/routes/grantform/GrantFormSuccess.js]: It is. Keeping it in the route is much nicer. Will make changes to highlight a solution.

**Review by kristjanmik — COMMENTED** [review]
Will commit some changes

File: src/routes/grantform/GrantForm.css [hunk 4]
```diff
@@ -15,14 +20,140 @@
...
+
+.formErrorMessage {
+  padding-top: 10px;
+  text-align: center;
+}
+
+/* .success {
+  width: 84px;
+  height: 40px;
+} */
```
Comment (johannth) [inline src/routes/grantform/GrantForm.css]: Lol, what a derp on my part. Must have made this error in the final stage when the linter was complaining about unused styles and I split the css into two files.

File: src/routes/grantform/GrantFormSuccess.js [hunk 5]
```diff
@@ -36,12 +28,12 @@ const GrantFormSuccess = ({ submission }) => {
...
 };
 GrantFormSuccess.propTypes = {
-  submission: PropTypes.object,
+  email: PropTypes.string,
 };
 
 export default connect(
   (state /* , props */) => ({
-    submission: state.grant.submission,
+    ...state.grant.submission,
```
Comment (johannth) [inline src/routes/grantform/GrantFormSuccess.js]: Nice change.

File: src/routes/grantform/index.js [hunk 6]
```diff
@@ -17,7 +17,13 @@ export default [
   {
     path: '/grant/success',
 
-    async action() {
+    async action({ store }) {
+      const { submission } = store.getState();
+
+      if (!submission) {
+        return { redirect: '/' };
+      }
```
Comment (johannth) [inline src/routes/grantform/index.js]: Ok nice. So each action here has access to the store. This is great. I was imagining that this was possible but couldn't see an example anywhere.

**Comment by johannth** [comment]
Nice additions. I'm merging this.

File: src/routes/grantform/index.js (hunk 6 above)
Comment (kristjanmik) [reply src/routes/grantform/index.js]: Yeah, there are plenty in there, you can log it out to see. My most used ones are 
```
{ store, userAgent, params }
```
```
path: '/admin/stuff/:id',
  async action({ store, userAgent, params }) {
    const { user } = store.getState();
    const { id } = params; //get the id from the url
...
```

**Pull request merged**